propertyestimator/tests/test_utils/test_openmm.py

```diff
@@ -0,0 +1,191 @@
+import inspect
+from random import randint, random
+
+import numpy as np
+import pytest
+from simtk import unit as simtk_unit
+
+from propertyestimator import unit
+from propertyestimator.utils.openmm import openmm_quantity_to_pint, pint_quantity_to_openmm, openmm_unit_to_pint, \
+    pint_unit_to_openmm, unsupported_openmm_units
+
+
+def _get_all_openmm_units():
+    """Returns a list of all of the defined OpenMM units.
+
+    Returns
+    -------
+    list of simtk.unit.Unit
+    """
+
+    all_openmm_units = set([unit_type for _, unit_type in inspect.getmembers(simtk_unit)
+                           if isinstance(unit_type, simtk_unit.Unit)])
+
+    all_openmm_units = all_openmm_units.difference(unsupported_openmm_units)
+
+    return all_openmm_units
+
+
+def _get_all_pint_units():
+    """Returns a list of all of the pint units which
+    could be converted from all OpenMM units.
+
+    Returns
+    -------
+    list of pint.Unit
+    """
+
+    all_openmm_units = _get_all_openmm_units()
+
+    all_pint_units = [openmm_unit_to_pint(openmm_unit) for openmm_unit in all_openmm_units]
+    return all_pint_units
+
+
+def test_daltons():
+
+    openmm_quantity = random() * simtk_unit.dalton
+    openmm_raw_value = openmm_quantity.value_in_unit(simtk_unit.gram / simtk_unit.mole)
+
+    pint_quantity = openmm_quantity_to_pint(openmm_quantity)
+    pint_raw_value = pint_quantity.to(unit.gram / unit.mole).magnitude
+
+    assert np.allclose(openmm_raw_value, pint_raw_value)
+
+
+@pytest.mark.parametrize("openmm_unit", _get_all_openmm_units())
+@pytest.mark.parametrize("value", [random(), randint(1, 10), [random(), random()], np.array([random(), random()])])
```

Comment:
This is interesting. If this is going to fail for particular values of `value`, does it make sense that we should see that failure during testing at a random time in the future (probably during unrelated development)? I think it would be better to have a --runslow option that tests a pretty exhaustive set of values, but during normal testing just have this regularly run with a fixed value. [1]

[1] The fixed value could still be random :-P https://www.xkcd.com/221/

Reply:
This should _never_ fail in practice. If it does - it means we've found a really weird edge case that would be good to know about and `random` allows us to be on the hunt for this (even though I think it will never find any issues)